wp-security-audit-log.php

```diff
@@ -27,612 +27,617 @@
+        foreach($wpdb->get_results($sql, ARRAY_A) as $item)
+            $events[$item['EventID']] = $item;
+        $sql = 'SELECT * FROM ' . $wpdb->base_prefix . 'wordpress_auditlog';
+        $auditlog = $wpdb->get_results($sql, ARRAY_A);
+        
+        // migrate using db logger
+        foreach($auditlog as $entry){
+            $data = array(
+                'ClientIP' => $entry['UserIP'],
+                'UserAgent' => '',
+                'CurrentUserID' => $entry['UserID'],
+            );
+            if($entry['UserName'])
+                $data['Username'] = base64_decode($entry['UserName']);
+            $mesg = $events[$entry['EventID']]['EventDescription'];
+            $date = strtotime($entry['EventDate']);
+            $type = $entry['EventID'];
+            if(isset($migTypes[$type]))$type = $migTypes[$type];
+            // convert message from '<strong>%s</strong>' to '%Arg1%' format
+            $c = 0; $n = '<strong>%s</strong>'; $l = strlen($n);
+            while(($pos = strpos($mesg, $n)) !== false){
+                $mesg = substr_replace($mesg, '%MigratedArg' . ($c++) .'%', $pos, $l);
+            }
+            $data['MigratedMesg'] = $mesg;
+            // generate new meta data args
+            $temp = unserialize(base64_decode($entry['EventData']));
+            foreach((array)$temp as $i => $item)
+                $data['MigratedArg' . $i] = $item;
+            // send event data to logger!
+            foreach($this->alerts->GetLoggers() as $logger){
+                $logger->Log($type, $data, $date, $entry['BlogId'], true);
+            }
+        }
+        
+        // migrate settings
+        $this->settings->SetAllowedPluginEditors(
+            get_option('WPPH_PLUGIN_ALLOW_CHANGE')
+        );
+        $this->settings->SetAllowedPluginViewers(
+            get_option('WPPH_PLUGIN_ALLOW_ACCESS')
+        );
+        $s = get_option('wpph_plugin_settings');
+        //$this->settings->SetPruningDate(($s->daysToKeep ? $s->daysToKeep : 30) . ' days');
+        //$this->settings->SetPruningLimit(min($s->eventsToKeep, 1));
+        $this->settings->SetViewPerPage(max($s->showEventsViewList, 5));
+        $this->settings->SetWidgetsEnabled(!!$s->showDW);
+    }
+    
+    // </editor-fold>
+    
+    // <editor-fold desc="Utility Methods">
+    
+    /**
+     * @return string The current plugin version (according to plugin file metadata).
+     */
+    public function GetNewVersion(){
+        $version = get_plugin_data(__FILE__, false, false);
+        return isset($version['Version']) ? $version['Version'] : '0.0.0';
+    }
+    
+    /**
+     * @return string The plugin version as stored in DB (will be the old version during an update/install).
+     */
+    public function GetOldVersion(){
+        return $this->GetGlobalOption('version', '0.0.0');
+    }
+    
+    /**
+     * @internal To be called in admin header for hiding plugin form Plugins list.
+     */
+    public function HidePlugin() {

```

Comment:
In the new wordpress is used the attribute data-slug instead of the id for identify each row